### PR TITLE
Story dfc 10506 nightly refresh of apprenticeships+courses

### PR DESCRIPTION
**Pull request by IanCrisp**

### CODE REVIEW

File: DFC.App.JobProfile.CurrentOpportunities/Controllers/SegmentController.cs [hunk 1]
```diff
@@ -69,6 +68,30 @@ public async Task<IActionResult> Document(string article)
...
+
+                logger.LogInformation($"{nameof(SimpleList)} has succeeded");
+            }
+            else
+            {
+                logger.LogWarning($"{nameof(SimpleList)} has returned with no results");
+            }
+
+            return this.NegotiateContentResult(viewModel, viewModel.Items);
+        }
```
Comment (Muthuramana) [inline DFC.App.JobProfile.CurrentOpportunities/Controllers/SegmentController.cs]: Do we need this? Can we not use the default index action and add content negotiation to that?

File: DFC.App.JobProfile.CurrentOpportunities/Views/Segment/SimpleList.cshtml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        <dl>
+            @foreach (var document in Model.Items)
+            {
+                <dt>@document.CanonicalName</dt>
+                <dd>@document.DocumentId</dd>
+            }
+        </dl>
+    }
+</body>
+</html>
```
Comment (Muthuramana) [inline DFC.App.JobProfile.CurrentOpportunities/Views/Segment/SimpleList.cshtml]: Do we need this? Very similar to the default view.

File: DFC.App.JobProfile.CurrentOpportunities.MessageFunctionApp/Functions/DeleteSegment.cs [hunk 3]
```diff
@@ -15,32 +15,32 @@ public static class DeleteSegment
...
+        //        if (result == HttpStatusCode.OK)
+        //        {
+        //            log.LogInformation($"{ThisClassPath}: JobProfile Id: {serviceBusModel.JobProfileId}: Deleted segment");
+        //        }
+        //        else
+        //        {
+        //            log.LogWarning($"{ThisClassPath}: JobProfile Id: {serviceBusModel.JobProfileId}: Segment not deleted: Status: {result}");
+        //        }
+        //    }
+        //}
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.JobProfile.CurrentOpportunities.MessageFunctionApp/Functions/DeleteSegment.cs]: removed as requested

File: DFC.App.JobProfile.CurrentOpportunities.MessageFunctionApp/Functions/PatchSegmentMarkup.cs [hunk 4]
```diff
@@ -16,43 +16,43 @@ public static class PatchSegmentMarkup
...
+        //        else
+        //        {
+        //            log.LogWarning($"{ThisClassPath}: JobProfile Id: {serviceBusModel.JobProfileId}: Segment not patched: Status: {result}");
+        //        }
+        //    }
+        //    else
+        //    {
+        //        log.LogWarning($"{ThisClassPath}: JobProfile Id: {serviceBusModel.JobProfileId}: Service bus message is stale: {serviceBusModel.Data.LastReviewed}, stored: {segmentModel.Data.LastReviewed}");
+        //    }
+        //}
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.JobProfile.CurrentOpportunities.MessageFunctionApp/Functions/PatchSegmentMarkup.cs]: removed as requested

File: DFC.App.JobProfile.CurrentOpportunities.MessageFunctionApp/Functions/PostSegment.cs [hunk 5]
```diff
@@ -15,40 +15,40 @@ public static class PostSegment
...
+        //        else
+        //        {
+        //            log.LogWarning($"{ThisClassPath}: JobProfile Id: {serviceBusModel.DocumentId}: Segment not Posted: Status: {result}");
+        //        }
+        //    }
+        //    else
+        //    {
+        //        log.LogWarning($"{ThisClassPath}: JobProfile Id: {serviceBusModel.DocumentId}: Service bus message is stale: {serviceBusModel.Data.LastReviewed}, stored: {segmentModel.Data.LastReviewed}");
+        //    }
+        //}
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.JobProfile.CurrentOpportunities.MessageFunctionApp/Functions/PostSegment.cs]: removed as requested

File: DFC.App.JobProfile.CurrentOpportunities.MessageFunctionApp/Services/HttpClientService.cs [hunk 6]
```diff
@@ -13,78 +13,74 @@ namespace DFC.App.JobProfile.CurrentOpportunities.MessageFunctionApp.Services
...
+        //    using (var request = new HttpRequestMessage(HttpMethod.Delete, url))
+        //    {
+        //        request.Headers.Accept.Clear();
+        //        request.Headers.Accept.Add(new System.Net.Http.Headers.MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json));
+
+        //        var response = await httpClient.SendAsync(request).ConfigureAwait(false);
+
+        //        return response.StatusCode;
+        //    }
+        //}
```
[the remark this answers is not in the document]
Comment (IanCrisp) [reply DFC.App.JobProfile.CurrentOpportunities.MessageFunctionApp/Services/HttpClientService.cs]: removed as requested

File: DFC.App.JobProfile.CurrentOpportunities/Controllers/SegmentController.cs (hunk 1 above)
Comment (IanCrisp) [reply DFC.App.JobProfile.CurrentOpportunities/Controllers/SegmentController.cs]: removed as requested, now using "Index" action instead

File: DFC.App.JobProfile.CurrentOpportunities/Views/Segment/SimpleList.cshtml (hunk 2 above)
Comment (IanCrisp) [reply DFC.App.JobProfile.CurrentOpportunities/Views/Segment/SimpleList.cshtml]: removed as requested, now using "Index" action instead

**Pull request merged**